modules/imaging_browser/js/imagingbrowser.js

```diff
@@ -1,28 +1,35 @@
-/*global document, $, window*/
 $(document).ready(function () {
     "use strict";
     var callback = function (extraparam) {
-        return function () {
-            var checked = $("input.mripanel:checkbox:checked"), i,
-                FileIDs = [];
-            console.log(checked);
 
-            if (extraparam === undefined) {
-                extraparam = '';
-            }
+        // Ensure extraparam is not undefined
+        extraparam = extraparam || '';
+
+        return function () {
+            var checked = $("input.mripanel:checkbox:checked");
+ 
+            if (0 < checked.size()) {
+                // Build the url params based on selected image(s)
+                var FileIDs = [];
+                checked.each(function (index, element) {
+                    FileIDs.push(element.dataset.fileId);
+                });
```

Comment:
This should be moved back into the for loop. Since the element is pushed to the FileIDs in the callback, there is a possibility that line 20 might be executed before this